### PR TITLE
refactor: Refactor Caller, optimize Caller to support identity authentication process

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

Refactor Caller, optimize Caller to support identity authentication process

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: WebApplication1/UserCaller.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
```
Comment (doddgu) [inline WebApplication1/UserCaller.cs:1]: ?

File: src/BuildingBlocks/Data/Masa.BuildingBlocks.Data.Contracts/Internal/ServiceScoped.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+[assembly: InternalsVisibleTo("Masa.BuildingBlocks.Data")]
+// ReSharper disable once CheckNamespace
+
+namespace Masa.BuildingBlocks.Data;
+
+public class ServiceScoped
```
Comment (doddgu) [inline src/BuildingBlocks/Data/Masa.BuildingBlocks.Data.Contracts/Internal/ServiceScoped.cs:9]: ServiceXXX? XXXService?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/BuildingBlocks/Data/Masa.BuildingBlocks.Data.Contracts/Internal/ServiceScoped.cs (hunk 2 above)
Comment (zhenlei520) [reply src/BuildingBlocks/Data/Masa.BuildingBlocks.Data.Contracts/Internal/ServiceScoped.cs:9]: The currently used format is: ServiceXXX

File: WebApplication1/UserCaller.cs (hunk 1 above)
Comment (zhenlei520) [reply WebApplication1/UserCaller.cs]: Processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**